src/proto/codec.rs

```diff
@@ -154,11 +154,14 @@ impl Decoder for WebSocketProtocol {
 
         let mask = if masked {
             offset += 4;
-            get_buf_if_space!(src, offset - 4..offset)
-                .try_into()
-                .unwrap()
+            // SAFETY: We know the slice is exactly 4 bytes long
+            unsafe {
+                &mut *(get_buf_if_space!(src, offset - 4..offset)
+                    .as_mut_ptr()
+                    .cast())
```

Comment:
I'm uncomfortable with this pointer to reference conversion with regard to aliasing. Could you get `(mask, payload)` with `split_at_mut` instead?